### PR TITLE
Added Google API Integration

### PR DESCRIPTION
**Pull request by jwfwessels**
* Added dataParser for googleSheets
* Added GoogleSheetsContext to replace useAppState hook
* replaced useage of useAppState with useContext

Note: missing tests, and updates to tests

### CODE REVIEW

File: src/utils/dataParser.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    if (date) {
+      return (dataRows as RowValues[]).map(row => flattenDataRows(row, keys));
+    } else {
+      // TODO: do something with other sheets like the config sheet.
+    }
+  }
+}
+
+function getSheetTableHeaders(row: KeyRowValues) {
+  // TODO: This should probably throw an error if the cell doesn't contain a string, Not sure how to Type that.
```
Comment (perlo27) [inline src/utils/dataParser.ts]: It probably makes sense to just use some default value in this case

File: src/utils/dataParser.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if (tempSheet) {
+      acc[sheet.properties.title] = tempSheet;
+    }
+    return acc;
+  }, {});
+}
+
+function flattenSheet(sheet: IncomingSheet) {
+  const dataRows = sheet.data[0].rowData;
+  const keys = getSheetTableHeaders(dataRows.shift()! as KeyRowValues);
```
Comment (perlo27) [inline src/utils/dataParser.ts]: `!` not needed
also i think it's a bit more readable and intuitive to extract values from dataRows with destructing syntax 
`const [keyRowValues, ...data] =  dataRows;
  const keys = getSheetTableHeaders(keyRowValues as KeyRowValues);
...
`;
overwise you would have to remember that you mutated dataRows using shift

File: src/utils/dataParser.ts (hunk 1 above)
Comment (jwfwessels) [reply src/utils/dataParser.ts]: well if the sheet doesn't have headers that's an error 😅. We the sheets to follow a convention

File: src/utils/dataParser.ts (hunk 2 above)
Comment (jwfwessels) [reply src/utils/dataParser.ts]: It complained about `!`  on my side as the sheet could be empty.

File: src/ContextProviders/GoogleSheetsContextProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React, { createContext, useState, useEffect } from 'react';
+import { ParseGoogleSheetsApiResponse } from 'src/utils/dataParser';
```
Comment (perlo27) [inline src/ContextProviders/GoogleSheetsContextProvider.tsx]: should this func be lowercase?

**Pull request merged**